tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot

```diff
@@ -1,26 +1,134 @@
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-178
+    Set Library Search Order    SeleniumLibrary
+    @{cur_usage} =    Read Current CPU Usage
+    Run Jupyter Notebook For 5 Minutes
+    @{after_running_program} =    Read Current CPU Usage
+    ${res} =    Checks Two Lists Equal Or Not    ${cur_usage}    ${after_running_program}
+    Should Be Equal    ${res}    False
+
+Test Metric "Rhods_Total_Users" On ODS Prometheus
+    [Documentation]    Verifies the total users are same as jupyter usres
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-628
+    ${expression} =    Set Variable    rhods_total_users&step=1
+    ${rhods_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${expression}
+    ${expression} =    Set Variable    jupyterhub_total_users&step=1
+    ${jupyterhub_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}
+    ...    ${expression}
+    Should Be Equal    ${rhods_total_users.json()["data"]["result"][0]["value"][-1]}
+    ...    ${jupyterhub_total_users.json()["data"]["result"][0]["value"][-1]}
 
+Test metric "rhods_aggregate_availability" on ODS Prometheus
+    [Documentation]    Verifies the rhods aggregate availability on rhods prometheus
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-636
+    ${expression} =    Set Variable    rhods_aggregate_availability&step=1
+    ${resp} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${expression}
+    Should Be Equal    ${resp.json()["data"]["result"][0]["value"][-1]}    1
+
+
+*** Keywords ***
 Check Prometheus Recording Rules
-  Prometheus.Verify Rules  ${RHODS_PROMETHEUS_URL}  ${RHODS_PROMETHEUS_TOKEN}  record  @{RECORD_GROUPS}
+    Prometheus.Verify Rules    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    record    @{RECORD_GROUPS}
 
 Check Prometheus Alerting Rules
-  Prometheus.Verify Rules  ${RHODS_PROMETHEUS_URL}  ${RHODS_PROMETHEUS_TOKEN}  alert  @{ALERT_GROUPS}
+    Prometheus.Verify Rules    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    alert    @{ALERT_GROUPS}
+
+Read Current CPU Usage
+    [Documentation]    Returns list of current cpu usage
+    ${version_check} =    Is RHODS Version Greater Or Equal Than    1.5.0
+    IF    ${version_check}==True
+        ${Expression} =    Set Variable
+        ...    sum(rate(container_cpu_usage_seconds_total{prometheus_replica="prometheus-k8s-0", container="",pod=~"jupyterhub-nb.*",namespace="rhods-notebooks"}[1h]))
+    ELSE
+        ${Expression} =    Set Variable
+        ...    sum(rate(container_cpu_usage_seconds_total{prometheus_replica="prometheus-k8s-0", container="",pod=~"jupyterhub-nb.*",namespace="rhods-notebooks"}[5m]))
+    END
+    ${resp} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${Expression}
+    [Return]    ${resp.json()["data"]["result"]}
+
+ Checks Two Lists Equal Or Not
```

Comment:
Not needed if using scalar variables to compare the values in the first test

There is a Lists Should Be Equal keyword in robotframework:   [Lists Should Be Equal](https://robotframework.org/robotframework/latest/libraries/Collections.html#Lists%20Should%20Be%20Equal)

Reply:
but it will fail, We have to check for ` list should not be equal `
that's why I wrote a keyword